aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -19,63 +19,64 @@
                              IntArrayRef stride, int64_t storage_offset,
                              MPSGraphTensor* inputTensor, const Tensor& self) {
   MPSGraphTensor *outputTensor = nil;
+  const size_t shape_size = size.size();
+
   @autoreleasepool {
-      int32_t* sizeArray = new int32_t[size.size()]();
-      for (int i = 0; i < size.size(); i++) {
-        sizeArray[i] = size[i];
+      int32_t* sizeArray = new int32_t[shape_size];
+      for (int i = 0; i < shape_size; i++) {
+        sizeArray[i] = static_cast<int32_t>(size[i]);
```

Comment:
Shouldn't we assert that this actually fits in `int32` and won't overflow?

Reply:
Good point, the support for shapes greater than what can fit in the int32 is not supported.